test/adapter/json_api/relationships_test.rb

```diff
@@ -158,10 +157,16 @@ def test_relationship_meta
           end
 
           def test_relationship_not_including_data
+            @author.define_singleton_method(:read_attribute_for_serialization) do |attr|
+              fail 'should not be called' if attr == :blog
```

Comment:
In any case, this test file is basically garbage. See  https://github.com/rails-api/active_model_serializers/pull/1857 would be better to do an 'integration' test